view-generator-framework/src/main/java/org/hypertrace/core/viewgenerator/service/ViewGeneratorLauncher.java

```diff
@@ -1,58 +1,108 @@
 package org.hypertrace.core.viewgenerator.service;
 
-import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
-import org.hypertrace.core.serviceframework.PlatformService;
+import static org.hypertrace.core.viewgenerator.service.ViewGeneratorConstants.INPUT_TOPIC_CONFIG_KEY;
+import static org.hypertrace.core.viewgenerator.service.ViewGeneratorConstants.KAFKA_STREAMS_CONFIG_KEY;
+import static org.hypertrace.core.viewgenerator.service.ViewGeneratorConstants.OUTPUT_TOPIC_CONFIG_KEY;
+import static org.hypertrace.core.viewgenerator.service.ViewGeneratorConstants.VIEW_GENERATOR_CLASS_CONFIG_KEY;
+
+import com.typesafe.config.Config;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.kstream.Consumed;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.Produced;
+import org.hypertrace.core.kafkastreams.framework.KafkaStreamsApp;
 import org.hypertrace.core.serviceframework.config.ConfigClient;
+import org.hypertrace.core.serviceframework.config.ConfigUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/**
- * ViewGeneratorLauncher reads enriched structured traces and converts them to Pinot records using specified Avro
- * schema specified in the target view generator class, which is specified in the config.
- */
-public class ViewGeneratorLauncher extends PlatformService {
-
-  private static Logger LOGGER = LoggerFactory.getLogger(ViewGeneratorLauncher.class);
+public class ViewGeneratorLauncher extends KafkaStreamsApp {
 
-  private ViewGenerationJob job;
-  private StreamExecutionEnvironment environment;
+  private static final Logger logger = LoggerFactory.getLogger(ViewGeneratorLauncher.class);
+  private static final String DEFAULT_VIEW_GEN_JOB_CONFIG_KEY = "view-gen";
+  private String viewGenName;
 
   public ViewGeneratorLauncher(ConfigClient configClient) {
     super(configClient);
   }
 
+  public String getViewGenName() {
```

Comment:
What are these get and set methods used for?

Reply:
It is used in `MultiViewGeneratorLauncher` where all view-generator are merged as a single KStream job. In that case,  getViewGenName is used from getJobConfigKey which help to fetch its specific config from properties.